pkg/diagnose/connectivity/trace_request.go

```diff
@@ -24,10 +24,65 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/resolver"
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
+func init() {
+	diagnosis.Register("connectivity-datadog-core-endpoints", diagnose)
+}
+
+func diagnose(diagCfg diagnosis.DiagnoseConfig) []diagnosis.Diagnosis {
+
+	// Create domain resolvers
+	keysPerDomain, err := utils.GetMultipleEndpoints(config.Datadog)
+	if err != nil {
+		return []diagnosis.Diagnosis{
+			{
+				Result:      diagnosis.DiagnosisSuccess,
+				Name:        "Endpoints configuration",
+				Diagnosis:   "Misconfiguration of agent endpoints",
+				Remediation: "Please validate Agent configuration",
+				RawError:    err,
+			},
+		}
+	}
+
+	var diagnoses []diagnosis.Diagnosis
+	domainResolvers := resolver.NewSingleDomainResolvers(keysPerDomain)
+	client := forwarder.NewHTTPClient(config.Datadog)
+
+	// Send requests to all endpoints for all domains
+	for _, domainResolver := range domainResolvers {
+		// Go through all API Keys of a domain and send an HTTP request on each endpoint
+		for _, apiKey := range domainResolver.GetAPIKeys() {
+			for _, endpointInfo := range endpointsInfo {
+				domain, _ := domainResolver.Resolve(endpointInfo.Endpoint)
+				ctx := context.Background()
+				statusCode, responseBody, logURL, err := sendHTTPRequestToEndpoint(ctx, client, domain, endpointInfo, apiKey)
+
+				// Check if there is a response and if it's valid
+				report, reportErr := verifyEndpointResponse(statusCode, responseBody, err)
+				d := diagnosis.Diagnosis{
+					Name: "Connectivity to " + logURL,
+				}
+				if reportErr == nil {
+					d.Result = diagnosis.DiagnosisSuccess
+					d.Diagnosis = fmt.Sprintf("Connectivity to `%s` is Ok\n%s", logURL, report)
+				} else {
+					d.Result = diagnosis.DiagnosisFail
+					d.Diagnosis = fmt.Sprintf("Connection to `%s` is falied\n%s", logURL, report)
+					d.Remediation = "Please validate Agent configuration and firewall to access " + logURL
+					d.RawError = err
+				}
+				diagnoses = append(diagnoses, d)
+			}
+		}
+	}
+	return diagnoses
+}
+
 // RunDatadogConnectivityDiagnose sends requests to all known endpoints for all domains
 // to check if there are connectivity issues between Datadog and these endpoints
 func RunDatadogConnectivityDiagnose(writer io.Writer, noTrace bool) error {
```

Comment:
That is one of my main question to you guys.  This is what I have said in https://github.com/DataDog/datadog-agent/pull/16719#discussion_r1184442552

> This is where is probably made a wrong decision from the very beginning. I have decided, afraid to change/break existing CLI, to keep "new" diagnose CLI interface separate from what we already have. I did not know exactly how "far" I will or should I go and certainly did not want to jeopardize of what is already there. I feel right now, and I have implied that in the PR comment, that I should not have "all" as segregated CLI command and instead integrate existing CLI commands into existing framework at implementation and at CLI level. In that case your suggestion would be even relevant in a sense that that would the only possible way to deal with that. May be I should take another swing into that (in a separate PR). Not sure if I explain it well though.

I can delay this PR merge and re-do CLI to eliminate "all" and other more specific sub-diagnoses but originally I was afraid to touch it because I did not know where and how I endup with the implementation.